### PR TITLE
Add comprehensive PR readiness prompt for AI agents

### PR DESCRIPTION
**Pull request by jcouball**
## Description

This PR adds a comprehensive "AI Prompt: Create Comprehensive PR" section to `.github/copilot-instructions.md` that provides a structured workflow for AI agents to verify PR readiness at the end of implementation.

## Motivation

Currently, PR readiness guidelines are scattered across multiple documents (CONTRIBUTING.md, copilot-instructions.md, PR template). This makes it easy to miss important quality checks. This new prompt consolidates and operationalizes these guidelines into one actionable checklist that AI agents can use to ensure nothing is missed before PR submission.

## What's Added

A new section with 7 verification steps:

1. **Run Final Validation** - Execute tests, rubocop, check for warnings
2. **Verify Testing Quality** - Emphasizes 100% unit coverage and minimal integration tests
3. **Review Code Quality** - YARD docs, security, cross-platform compatibility
4. **Verify Against Git Documentation** - Ensure all edge cases from git-scm.com are tested
5. **Check Commit Quality** - Conventional Commits format validation
6. **Review Documentation** - Architecture docs, README, YARD accuracy
7. **Generate PR Summary** - Comprehensive report with implementation details, test coverage, and suggested PR materials

## Benefits

- **Proactive quality assurance** - Catches issues before CI runs
- **Consistent standards** - Operationalizes the project's high bar for testing
- **Faster feedback** - Local validation is faster than CI revision cycles
- **Better PRs** - Comprehensive summary generation improves PR descriptions

## Checklist

- [x] I reviewed and applied the project's [AI Policy](../AI_POLICY.md). I understand and verify any AI-assisted changes included in this PR and ensured they meet quality, security, and licensing standards.
- [x] Tests added/updated as needed and `rake` passes locally.
- [x] Documentation updated where applicable (README and/or YARD).

### CODE REVIEW

File: .github/copilot-instructions.md [hunk 1]
```diff
@@ -2861,6 +2861,101 @@ description" --body "..."`
...
+Use this at the end of implementation to prepare for PR submission:
+
+**"I've completed the implementation. Please perform a comprehensive PR readiness review."**
+
+### 1. Run Final Validation
+
+Execute and report results for:
+- `bundle exec rake test_all` - all RSpec and Test::Unit tests must pass
+- `bundle exec rubocop` on all changed files - zero violations required
+- Check test output for any Ruby warnings
```
[the remark this answers is not in the document]
Comment (jcouball) [reply .github/copilot-instructions.md:2875]: Explicitly include a `yard` step.
